src/plonk/prover.rs

```diff
@@ -529,50 +529,70 @@ impl<C: CurveAffine> Proof<C> {
 
-        let x_6: C::Scalar = get_challenge_scalar(Challenge(transcript.squeeze().get_lower_128()));
+            let mut f_blind = Blind(C::Scalar::random());
+            let f_commitment = params.commit(&f_poly_dup, f_blind).to_affine();
+            hash_point(&mut transcript_dup, &f_commitment)?;
 
-        let mut q_evals = vec![C::Scalar::zero(); meta.rotations.len()];
+            let x_6: C::Scalar =
+                get_challenge_scalar(Challenge(transcript_dup.squeeze().get_lower_128()));
 
-        for (_, &point_index) in meta.rotations.iter() {
-            q_evals[point_index.0] =
-                eval_polynomial(&q_polys[point_index.0].as_ref().unwrap(), x_6);
-        }
+            let mut q_evals = vec![C::Scalar::zero(); meta.rotations.len()];
 
-        for eval in q_evals.iter() {
-            transcript_scalar.absorb(*eval);
-        }
+            for (_, &point_index) in meta.rotations.iter() {
+                q_evals[point_index.0] =
+                    eval_polynomial(&q_polys[point_index.0].as_ref().unwrap(), x_6);
+            }
 
-        let transcript_scalar_point =
-            C::Base::from_bytes(&(transcript_scalar.squeeze()).to_bytes()).unwrap();
-        transcript.absorb(transcript_scalar_point);
+            for eval in q_evals.iter() {
+                transcript_scalar_dup.absorb(*eval);
+            }
 
-        let x_7: C::Scalar = get_challenge_scalar(Challenge(transcript.squeeze().get_lower_128()));
+            let transcript_scalar_point =
+                C::Base::from_bytes(&(transcript_scalar_dup.squeeze()).to_bytes()).unwrap();
+            transcript_dup.absorb(transcript_scalar_point);
 
-        for (_, &point_index) in meta.rotations.iter() {
-            f_blind *= x_7;
-            f_blind += q_blinds[point_index.0];
+            let x_7: C::Scalar =
+                get_challenge_scalar(Challenge(transcript_dup.squeeze().get_lower_128()));
 
-            parallelize(&mut f_poly, |f, start| {
-                for (f, a) in f
-                    .iter_mut()
-                    .zip(q_polys[point_index.0].as_ref().unwrap()[start..].iter())
-                {
-                    *f *= &x_7;
-                    *f += a;
-                }
-            });
-        }
+            for (_, &point_index) in meta.rotations.iter() {
+                f_blind *= x_7;
+                f_blind += q_blinds[point_index.0];
+
+                parallelize(&mut f_poly_dup, |f, start| {
+                    for (f, a) in f
+                        .iter_mut()
+                        .zip(q_polys[point_index.0].as_ref().unwrap()[start..].iter())
+                    {
+                        *f *= &x_7;
+                        *f += a;
+                    }
+                });
+            }
 
-        // Let's prove that the q_commitment opens at x to the expected value.
-        let opening = OpeningProof::create(&params, &mut transcript, &f_poly, f_blind, x_6)
```

Comment:
I would recommend instead that we have `OpeningProof` itself fail, and then recover from that case to resume the `loop { }`. Otherwise, we're leaking implementation details of `OpeningProof` into the unrelated PLONK prover code.